manifests/config.pp

```diff
@@ -7,10 +6,28 @@
 #
 class mosquitto::config (
   Array[String[1]] $config = $mosquitto::config,
+  $prefix                  = undef
 ) {
   assert_private()
+  # Sensible defaults for the config prefix if none is specified
+  unless $prefix{
```

Comment:
I'd expect some if/else where the prefix is used. Now it's just behaving as a boolean.